src/middleware/queriesV2.ts

```diff
@@ -0,0 +1,117 @@
+import gql from 'graphql-tag';
+
+import { QueryDidsWithClaimsArgs } from '~/middleware/types';
+import { GraphqlQuery } from '~/types/internal';
+
+/**
+ *  @hidden
+ */
+function createClaimsFilters(variables: QueryDidsWithClaimsArgs): {
+  args: string[];
+  filters: string[];
+} {
+  const args = [];
+  const filters = [];
+  const { dids, claimTypes, trustedClaimIssuers, scope, includeExpired } = variables;
+  if (dids) {
+    args.push('$dids: [String!]');
+    filters.push('targetId: { in: $dids }');
```

Comment:
In previous middleware POC, @polymath-eric you pointed out that Apollo client would handle the part but apparently the `filter` can neither be empty not can have empty objects. Hence, we need to handle this separately